src/sliceview/jxl/src/lib.rs

```diff
@@ -62,22 +50,10 @@ pub fn height(ptr: *mut u8, input_size: usize, output_size: usize) -> i32 {
         slice::from_raw_parts(ptr, input_size)
     };
 
-    let image = match JxlImage::builder().read(data) {
-        Ok(image) => image,
-        Err(_image) => return -2,
-    };
-
-    for keyframe_idx in 0..image.num_loaded_keyframes() {
-        let frame = match image.render_frame(keyframe_idx) {
-            Ok(frame) => frame,
-            Err(_frame) => return -3,
-        };
-
-        let stream = frame.stream();
-        return stream.height() as i32;
+    match JxlImage::builder().read(data) {
```

Comment:
This is definitely a big improvement.  We could also avoid decoding the header multiple times by writing width and height to a *mut u8 buffer provided by the caller since we can only return a single value.

Reply:
I think that does actually work --- I think u64 and i64 correspond to bigint in javascript